### PR TITLE
Support Trio out-of-the-box

### PR DESCRIPTION
**Pull request by njsmith**
This PR makes `@retry` just work when running under Trio.

For now, I added `trio` to the test deps in tox.ini, so we can test
it. The code is also designed to handle things gracefully if trio
isn't installed, but currently there's not any automated test for
that... how would you like to handle this? Maybe add another
test environment without trio?

### CODE REVIEW

**Review by jd — DISMISSED** [review]
Yeah I think we should cover both. Just add a new tox scenario with and without trio :)

**Comment by njsmith** [comment]
@jd How's this look?

**Review by jd — CHANGES_REQUESTED** [review]
I think it's good though I'd prefer to have `py39-trio` rather than `notrio` 😁

**Comment by njsmith** [comment]
Fair enough!

**Comment by njsmith** [comment]
Made that change and also rebased on latest master, to make mergify happy

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,6 +64,15 @@ jobs:
...
             tox -e py39
+  py39-notrio:
+    docker:
+      - image: circleci/python:3.9
+    steps:
+      - checkout
+      - run:
+          command: |
+            sudo pip install tox
+            tox -e py39-notrio
```
Comment (penguinolog) [inline .circleci/config.yml:75]: inconsistent with `tox.int` (`py39-trio` for `trio` and normal python versions for other)

**Comment by chrisjwelly** [comment]
Hello @njsmith ! I understand it has been over 2 years since this PR was opened. Might there be any chance of a follow-up to support trio out of the box? 

I think this feature is great, as it can be easy to forget add `trio.sleep` argument to the retry.

**Comment by SysuJayce** [comment]
I also wonder how to use tenacity with trio.

If I use tenacity.retry() as a decorator for a async function, how can I check the result of the async function via `tenacity.retry_if_result()`? And then I got an error `TypeError: argument of type 'coroutine' is not iterable, xxx was never awaited`.

Even if I change my retry condition to `retry_if_exception`, I got another error `TypeError: object NoneType can't be used in 'await' expression`.

Please give me some tips, thank you guys~~~

**Comment by jakkdl** [comment]
This can be closed, as it was implemented with #463

**Pull request closed without merging**